pkg/routeagent/controllers/route/route.go

```diff
@@ -792,15 +792,15 @@ func (r *Controller) cleanVxSubmarinerRoutes() {
 		klog.Errorf("Unable to cleanup routes, error retrieving routes on the link %s: %v", VxLANIface, err)
 		return
 	}
-	for _, route := range currentRouteList {
-		klog.V(log.DEBUG).Infof("Processing route %v", route)
-		if route.Dst == nil || route.Gw == nil {
+	for i := range currentRouteList {
+		klog.V(log.DEBUG).Infof("Processing route %v", currentRouteList[i])
+		if currentRouteList[i].Dst == nil || currentRouteList[i].Gw == nil {
 			klog.V(log.DEBUG).Infof("Found nil gw or dst")
 		} else {
-			if r.remoteSubnets.Contains(route.Dst.String()) {
-				klog.V(log.DEBUG).Infof("Removing route %s", route.String())
-				if err = netlink.RouteDel(&route); err != nil {
-					klog.Errorf("Error removing route %s: %v", route.String(), err)
+			if r.remoteSubnets.Contains(currentRouteList[i].Dst.String()) {
+				klog.V(log.DEBUG).Infof("Removing route %s", currentRouteList[i].String())
```

Comment:
```suggestion
				klog.V(log.DEBUG).Infof("Removing route %s", currentRouteList[i])
```

I know the call to `String` was pre-existing but it would be nice to remove it since you already touched the line 😄 Also on the other lines below.

Reply:
Done!